agent/api/task/service_connect.go

```diff
@@ -65,3 +81,246 @@ type DNSConfigEntry struct {
+			}
+			if err := json.Unmarshal([]byte(data), scConfigValue); err != nil {
+				return nil, fmt.Errorf("failed to unmarshal attachment property value which contains service connect config value: %w", err)
+			}
+		case serviceConnectContainerNameKey:
+			// extract service connect container name from attachment properties
+			containerName = aws.StringValue(property.Value)
+		default:
+			return nil, fmt.Errorf("received an unrecognized attachment property: %s", property.String())
+		}
+	}
+	scConfigValue.ContainerName = containerName
+	if err := ValidateSCConfig(scConfigValue, taskContainers, networkMode); err != nil {
+		return nil, fmt.Errorf("service connect config validation failed: %w", err)
+	}
+	return scConfigValue, nil
+}
+
+// validateDnsConfig validates the service connnect dns config exists if egress config exists.
+func validateDnsConfig(scDnsConfligList []DNSConfigEntry, scEgressConfig EgressConfig) error {
+	// DNSConfig will also be empty as EgressConfig can be empty for the first service
+	// since there are no other tasks that it can talk to
+	if len(scDnsConfligList) < 1 {
+		if scEgressConfig.ListenerName != "" {
+			return fmt.Errorf("service connect config validation: no service connect DNS config")
+		}
+	}
+	for _, dnsEntry := range scDnsConfligList {
+		// HostName is a required field
+		if dnsEntry.HostName == "" {
+			return fmt.Errorf("service connect config validation: no service connect DNS config host name")
+		}
+		// Address is a required field
+		if dnsEntry.Address == "" {
+			return fmt.Errorf("service connect config validation: no service connect DNS config address")
+		}
+	}
+	return nil
+}
+
+// validateStatsConfig validates the service connect stats config path exists.
+func validateStatsConfig(scStatsConfig StatsConfig) error {
+	// stats config is a required field
+	if scStatsConfig.Path == "" {
+		return fmt.Errorf("service connect config validation: no service connect stats config")
+	}
+	return nil
+}
+
+// validateContainerName validates the service connect container name exists, and it exists in the task definition as well.
+func validateContainerName(scContainerName string, taskContainers []*ecsacs.Container) error {
+	// container name is a required field
+	if scContainerName == "" {
+		return fmt.Errorf("service connect config validation: no service connect container name")
+	}
+	// validate the specified service connect container name exists in task
+	for _, container := range taskContainers {
+		if aws.StringValue(container.Name) == scContainerName {
+			return nil
+		}
+	}
+	return fmt.Errorf("service connect config validation: service connect container name: %s does not exist in task", scContainerName)
+}
+
+// validateCIDR validates the passed CIDR is valid.
+func validateCIDR(cidr, version string) error {
+	_, _, err := net.ParseCIDR(cidr)
+	if err != nil {
+		return fmt.Errorf("service connect config validation: %s: %s is invalid", version, cidr)
```

Comment:
Could we also add a validation on the version?

Reply:
`validateCIDR` is added in commit 3d8d95406e74fc25d3c00d384a11cd5cd4ae55f9. Please take a look on `service_connect_validation.go` and `service_connect_validation_test.go`. Thanks.